include/ensmallen_bits/sdp/primal_dual_impl.hpp

```diff
@@ -33,57 +33,16 @@
 #include "lin_alg.hpp"
 
 namespace ens {
-
-template<typename DeprecatedSDPType>
-ens_deprecated
-PrimalDualSolver<DeprecatedSDPType>::PrimalDualSolver(
-    const DeprecatedSDPType& sdp) :
-    deprecatedSDP(sdp),
-    initialX(arma::eye<arma::mat>(sdp.N(), sdp.N())),
-    initialYSparse(arma::ones<arma::vec>(sdp.NumSparseConstraints())),
-    initialYDense(arma::ones<arma::vec>(sdp.NumDenseConstraints())),
-    initialZ(arma::eye<arma::mat>(sdp.N(), sdp.N())),
-    maxIterations(1000),
-    tau(0.99),
-    normXzTol(1e-7),
-    primalInfeasTol(1e-7),
-    dualInfeasTol(1e-7)
-{ /* Nothing to do. */ }
-
-template<typename DeprecatedSDPType>
-ens_deprecated
-PrimalDualSolver<DeprecatedSDPType>::PrimalDualSolver(
-    const DeprecatedSDPType& sdp,
-    const arma::mat& initialX,
-    const arma::vec& initialYSparse,
-    const arma::vec& initialYDense,
-    const arma::mat& initialZ) :
-    deprecatedSDP(sdp),
-    initialX(initialX),
-    initialYSparse(initialYSparse),
-    initialYDense(initialYDense),
-    initialZ(initialZ),
-    maxIterations(1000),
-    tau(0.99),
-    normXzTol(1e-7),
-    primalInfeasTol(1e-7),
-    dualInfeasTol(1e-7)
-{
-  // Nothing to do.
-}
-
-template<typename DeprecatedSDPType>
-PrimalDualSolver<DeprecatedSDPType>::PrimalDualSolver(
-    const size_t maxIterations,
-    const double tau,
-    const double normXzTol,
-    const double primalInfeasTol,
-    const double dualInfeasTol) :
-    maxIterations(maxIterations),
-    tau(tau),
-    normXzTol(normXzTol),
-    primalInfeasTol(primalInfeasTol),
-    dualInfeasTol(dualInfeasTol)
+inline PrimalDualSolver::PrimalDualSolver(const size_t maxIterations,
+                                          const double tau,
+                                          const double normXzTol,
+                                          const double primalInfeasTol,
+                                          const double dualInfeasTol) :
+                                          maxIterations(maxIterations),
```

Comment:
One last thing before we merge, the initialization list should be double tabbed, like you can see [here](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#constructor-initialization-lists)

Reply:
@favre49 I can just make these changes during merge (or alternately you can merge and make the fixes too), up to you. :+1:  (I should also get the style checks running on this repository too...)

Reply:
Oh alright, I wasn't aware. So I just commit directly to the contributor's branch? Or is there a GitHub way to do this?

Reply:
Yeah, I think you can do that; what I typically do is to follow the command-line instructions for how to merge from Github:

```
git checkout -b jwdinius-remove-sdp-depr master
git pull https://github.com/jwdinius/ensmallen.git remove-sdp-depr
```

And after this first step, I'll make the changes and commit (on the local `remove-sdp-depr` branch).  The rest of the process is just like in the Github guide:

```
git checkout master
git merge --no-ff jwdinius-remove-sdp-depr
git push origin master
```

Anyway, that's just what I do, there are probably about a billion other ways to do it too. :)

Reply:
Thanks, I'll try this

Reply:
Hmmm it blocks me from pushing to master saying 2 approving reviews are needed, unsure why.

Reply:
That's quite strange, I see this:
![reviewed](https://user-images.githubusercontent.com/1845039/77681671-12283f00-6f6c-11ea-88fa-7cc239c0116a.png)

Do you see something different?

Reply:
I see the same, not sure why I can't merge from command line with your instructions. Unless you meant that I should be handling this post-merge?

Reply:
What was the output when you tried to push from the command-line?  Maybe I have the permissions set up wrong or something with the protected master branch?

Reply:
```
remote: error: GH006: Protected branch update failed for refs/heads/master.
remote: error: At least 2 approving reviews are required by reviewers with write access.
To https://github.com/mlpack/ensmallen.git
 ! [remote rejected] master -> master (protected branch hook declined)
error: failed to push some refs to 'https://github.com/mlpack/ensmallen.git'
```

Reply:
Huh, I did some reading, and I think what's going on is this: https://github.community/t5/How-to-use-Git-and-GitHub/Pushing-to-a-protected-branch-with-required-status-checks/td-p/15966

Basically since the HEAD of the branch is changed after the local commits, I guess Github rejects the commit.  What happens if you merge the PR, then try to push a commit to master with the fixes afterwards?  (I'd like to debug this, because it's probably also the reason that other contributors haven't had success with the release scripts.  If you're up for trying to run the release script too, we can get the whole thing debugged. :))

Reply:
I'll give that a shot, I have the time now

Reply:
Very odd, I still get the same error